invoke/tasks.py

```diff
@@ -116,6 +116,38 @@ def __call__(self, *args, **kwargs):
     def called(self):
         return self.times_called > 0
 
+    def get_arg_names_and_spec_dict(self, spec):
+        arg_names = spec.args[:]
+        matched_args = [reversed(x) for x in [spec.args, spec.defaults or []]]
+        spec_dict = dict(zip_longest(*matched_args, fillvalue=NO_DEFAULT))
+        return arg_names, spec_dict
+
+    def get_undecorated_func(self, func):
+        """
+        If func is a decorated function, it returns the base undecorated
+        function. Otherwise, it returns None
+        """
+        if func.func_closure:
+            try:
+                # In the vast majority of circumstances, the first function
+                # in the list of closures is a function that func is decorating
+                undecorated_func = next(
+                    cell.cell_contents
+                    for cell
+                    in func.func_closure
+                    if inspect.isfunction(cell.cell_contents)
+                )
+
+                # If the undecorated function is a decorator itself, we want to recurse
```

Comment:
Yes, we lint things nowadays with flake8 (it should be in our `dev-requirements.txt` at this point), so please make sure anything `flake8` complains about gets fixed. It should see our `setup.cfg` and thus use the same rules that the Imhotep bot, or my own local run of flake8, are using.